### PR TITLE
DB-10328: clean up serialization for spark execution

### PR DESCRIPTION
**Pull request by jyuanca**

### CODE REVIEW

**Comment by jyuanca** [comment]
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

**Comment by spliceci** [comment]
TEST FAILED
http://nexus.splicemachine.com:8080/job/spliceengine-PR/job/Splice-PR-Branch/1742/

**Comment by jyuanca** [comment]
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

**Comment by spliceci** [comment]
TEST FAILED
http://nexus.splicemachine.com:8080/job/spliceengine-PR/job/Splice-PR-Branch/1755/

**Comment by jyuanca** [comment]
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

**Comment by spliceci** [comment]
TEST FAILED
http://nexus.splicemachine.com:8080/job/spliceengine-PR/job/Splice-PR-Branch/1757/

**Comment by jyuanca** [comment]
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

**Comment by spliceci** [comment]
TEST FAILED
http://nexus.splicemachine.com:8080/job/spliceengine-PR/job/Splice-PR-Branch/1770/

**Comment by jyuanca** [comment]
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

**Comment by spliceci** [comment]
TEST SUCCEEDED +1

**Review by msirek — APPROVED** [review]

**Comment by martinrupp** [comment]
have we checked this doesn't introduce upgrade problems like 10566?

**Comment by martinrupp** [comment]
> have we checked this doesn't introduce upgrade problems like 10566?

just checked, looks good!

File: splice_machine/src/test/java/com/splicemachine/triggers/Trigger_Referencing_Clause_IT.java [hunk 1]
```diff
@@ -887,6 +887,7 @@ public void testNestedTriggers() throws Exception {
 
     // Cause trigger rows to go over 1000000, causing a
     // rollback and switch to spark execution.
+    @Ignore
```
Comment (ipraznik-splice) [inline splice_machine/src/test/java/com/splicemachine/triggers/Trigger_Referencing_Clause_IT.java:890]: Should this test stay ignored? Is there a problem with test?

**Review by ipraznik-splice — CHANGES_REQUESTED** [review]
Test is set to "Ignore". Is it correct?

File: splice_machine/src/test/java/com/splicemachine/triggers/Trigger_Referencing_Clause_IT.java (hunk 1 above)
Comment (martinrupp) [reply splice_machine/src/test/java/com/splicemachine/triggers/Trigger_Referencing_Clause_IT.java:890]: the whole test class is still ignored. there's a couple of jiras:
- https://splicemachine.atlassian.net/browse/DB-9614
- https://splicemachine.atlassian.net/browse/DB-10539
- https://splicemachine.atlassian.net/browse/DB-10579

we're not fixing any of this here, right?

Comment (jyuanca) [reply splice_machine/src/test/java/com/splicemachine/triggers/Trigger_Referencing_Clause_IT.java:890]: We will need more work to fix these ITs

**Comment by jyuanca** [comment]
@ipraznik-splice I only fixed some regressions, and will leave the existing trigger issues for other jiras to address

**Review by ipraznik-splice — APPROVED** [review]

File: splice_machine/src/test/java/com/splicemachine/triggers/Trigger_Referencing_Clause_IT.java (hunk 1 above)
Comment (jpanko1) [reply splice_machine/src/test/java/com/splicemachine/triggers/Trigger_Referencing_Clause_IT.java:890]: To fix and re-enable these ITs, added https://splicemachine.atlassian.net/browse/DB-10616

**Review by carolp-503 — APPROVED** [review]
+1

**Pull request merged**